src/rules/camelcase.rs

```diff
@@ -0,0 +1,744 @@
+    self.check_ident(local, IdentToCheck::variable(local));
+    import_star_as_specifier.visit_children_with(self);
+  }
+
+  fn visit_export_namespace_specifier(
+    &mut self,
+    export_namespace_specifier: &ExportNamespaceSpecifier,
+    _: &dyn Node,
+  ) {
+    let ExportNamespaceSpecifier { name, .. } = export_namespace_specifier;
+    self.check_ident(name, IdentToCheck::variable(name));
+    export_namespace_specifier.visit_children_with(self);
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+  use crate::test_util::*;
+
+  #[test]
+  fn test_is_underscored() {
+    let tests = [
+      ("foo_bar", true),
+      ("fooBar", false),
+      ("FooBar", false),
+      ("foo_bar_baz", true),
+      ("_foo_bar_baz", true),
+      ("__foo_bar_baz__", true),
+      ("__fooBar_baz__", true),
+      ("__fooBarBaz__", false),
+      ("Sha3_224", true),
+      ("SHA3_224", false),
+    ];
+
+    for &(input, expected) in tests.iter() {
+      assert_eq!(expected, is_underscored(input));
+    }
+  }
+
+  #[test]
+  fn test_to_camelcase() {
+    let tests = [
+      ("foo_bar", "fooBar"),
+      ("fooBar", "fooBar"),
+      ("FooBar", "FooBar"),
+      ("foo_bar_baz", "fooBarBaz"),
+      ("_foo_bar_baz", "_fooBarBaz"),
+      ("__foo_bar_baz__", "__fooBarBaz__"),
+      ("Sha3_224", "SHA3_224"),
+      ("SHA3_224", "SHA3_224"),
+      ("_leading", "_leading"),
+      ("trailing_", "trailing_"),
+      ("_bothEnds_", "_bothEnds_"),
+    ];
+
+    for &(input, expected) in tests.iter() {
+      assert_eq!(expected, to_camelcase(input));
+    }
+  }
+
+  #[test]
+  fn test_to_hint() {
+    fn s(s: &str) -> String {
+      s.to_string()
+    }
+
+    let tests = [
+      (
+        IdentToCheck::Variable(s("foo_bar")),
+        "Consider renaming `foo_bar` to `fooBar`",
+      ),
+      (
+        IdentToCheck::Function(s("foo_bar")),
+        "Consider renaming `foo_bar` to `fooBar`",
+      ),
+      (
+        IdentToCheck::Class(s("foo_bar")),
+        "Consider renaming `foo_bar` to `FooBar`",
+      ),
+      (
+        IdentToCheck::ObjectPat {
+          key_name: s("foo_bar"),
+          value_name: None,
+        },
+        "Consider replacing `{ foo_bar }` with `{ foo_bar: fooBar }`",
+      ),
+      (
+        IdentToCheck::ObjectPat {
+          key_name: s("foo_bar"),
+          value_name: Some(s("snake_case")),
+        },
+        "Consider renaming `snake_case` to `snakeCase`",
+      ),
+      (
+        IdentToCheck::NamedImport {
+          local: s("foo_bar"),
+          imported: None,
+        },
+        "Consider replacing `{ foo_bar }` with `{ foo_bar as fooBar }`",
+      ),
+      (
+        IdentToCheck::NamedImport {
+          local: s("foo_bar"),
+          imported: Some(s("snake_case")),
+        },
+        "Consider renaming `foo_bar` to `fooBar`",
+      ),
+    ];
+
+    for (error_ident, expected) in tests.iter() {
+      assert_eq!(*expected, error_ident.to_hint());
+    }
+  }
```

Comment:
I have implemented the functionality of emitting hints as a part of diagnostics.
As these test cases show, it emits the hints as follows:

```typescript
let foo_bar = 42;
function foo_bar() {}
const { snake_case: foo_bar = 42 } = obj;
import { snake_case as foo_bar } from 'mod.ts';
import * as foo_bar from 'mod.ts';
// For the above 5 cases, the hint will be like "Consider renaming `foo_bar` to `fooBar`"

class foo_bar {}
// the hint will be like "Consider renaming `foo_bar` to `FooBar`"
// NOTE: it's in pascal case

const { foo_bar } = obj;
// the hint will be like "Consider replacing `{ foo_bar }` with `{ foo_bar: fooBar }`"

import { foo_bar } from 'mod.ts';
// the hint will be like "Consider replacing `{ foo_bar }` with `{ foo_bar as fooBar }`"
import 


```

Reply:
@magurotuna make sure to pass those hints using new `Context::add_diagnostic_with_hint` which I added in #397

Reply:
Yes, I did already 👍 
https://github.com/denoland/deno_lint/blob/6b54ef5a2ecf0c3b5f4a0fd04fba8007faa86719/src/rules/camelcase.rs#L233-L243